### ISSUE TITLE
Angular in AOT mode

### ISSUE THREAD

**Issue by andreev-artem** [storybookjs/storybook#10760]
**Is your feature request related to a problem? Please describe.**
While developers work on Angular components in JIT mode it's possibly to get additional errors in AOT mode.

**Describe the solution you'd like**
We need a way to run Angular components in AOT mode while developing.

**Comment by shilman** [comment]
@kroeder any thoughts on thiss?

**Comment by kroeder** [comment]
Yes, see https://github.com/storybookjs/storybook/issues/10580#issuecomment-643577649

It's not a duplicate but related. AOT is something that needs to be supported and I'm experimenting with the AngularCompilerPlugin webpack plugin in order to achieve this. So far I was not able to produce AOT code

**Comment by stupidawesome** [comment]
I use code that depends on Ivy view engine being enabled, so this is a real blocker for me.

Looking at the current implementation, integration with storybook relies heavily on JIT compilation with dynamic calls to `NgModule` and `Component` decorators at runtime. AFAIK this won't work with AoT compilation. Ivy is also much stricter than View Engine here.

AoT compilation has [many restrictions](https://angular.io/guide/aot-compiler) to consider.

It should be possible to transform CSF stories into AoT compatible code.

Example:
```ts
export const Icon = () => ({
    moduleMetadata: {
        imports: [CommonModule],
        declarations: [AppIcon]
    },
    template: `
        <app-icon *ngFor="let icon of icons" [icon]="icon"></app-icon>
    `,
    props: {
        icons
    }
})
```

should generate something like

```ts
@Component({
    selector: "storybook-host",
    template: `
        <app-icon *ngFor="let icon of icons" [icon]="icon"></app-icon>
    `
})
export class CodegenHost {
    icons = icons
}

@NgModule({
    imports: [CommonModule],
    declarations: [AppIcon, CodegenHost],
    bootstrap: [CodegenHost]
})
export class Icon {}

platformBrowserDynamic.bootstrap(Icon)
```

Then add these as entry points for ngc to compile.

**Comment by kroeder** [comment]
@stupidawesome good to know, thanks! I think this will help me 

I haven't really started (just tried a first working version) 
Are you able to help me with https://github.com/storybookjs/storybook/pull/11157 ?
